# DataFrame Analysis Exercise

**Import the `bestsellers.csv` dataset and use it to answer the following questions:**

* Find the lowest User Rating in the DF
* Find the highest Price in the DF
* What is the average User Rating?
* What the average User Rating of the first 5 rows?
* What User Review score appeared the most?
* What is the total (sum) of all the values in the Reviews column?
* How many different authors are featured in the dataset?
* Which author wrote the most number of books on the list?  How many did they write?

In [2]:
import pandas as pd

In [4]:
bestsellers = pd.read_csv('data/bestsellers.csv')
bestsellers

Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  
0                    JJ Smith          4.7    17350      8  2016  Non Fiction  
1                Stephen King          4.6     2052     22  2011      Fiction  
2          Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3               George Orwell          4.7    21424      6  2017      Fiction  
4    National Geographic Kids          4.8     7665     12  2019  Non Fiction  
..                        ...          ...      ...    ...   ...          ...  
545               Jeff Kinney          4.9     9413      8  2019      Fiction  
546               Jen Sincero          4.7    14331      8  2016  Non Fiction  
547               Jen Sincero          4.7    14331      8  2017  Non Fiction  
548               Jen Sincero          4.7    14331      8  2018  Non Fiction  
549               Jen Sincero          4.7    14331      8  2019  Non Fiction  

[550 rows x 7 columns]

In [7]:
min_user = bestsellers.min()
min_user

Name           10-Day Green Smoothie Cleanse
Author                      Abraham Verghese
User Rating                              3.3
Reviews                                   37
Price                                      0
Year                                    2009
Genre                                Fiction
dtype: object

#### Find the lowest User Rating in the DF
* answer: 3.3

In [8]:
max_price = bestsellers.max()
max_price

Name           You Are a Badass: How to Stop Doubting Your Gr...
Author                                              Zhi Gang Sha
User Rating                                                  4.9
Reviews                                                    87841
Price                                                        105
Year                                                        2019
Genre                                                Non Fiction
dtype: object

#### Find the highest Price in the DF
* answer: 105

In [18]:
avg_user = bestsellers.mean(numeric_only=True)
avg_user

User Rating        4.618364
Reviews        11953.281818
Price             13.100000
Year            2014.000000
dtype: float64

#### What is the average User Rating?
* answer: 4.618364

In [19]:
avg_user5 = bestsellers.head().mean(numeric_only=True)
avg_user5

User Rating        4.7
Reviews        13494.0
Price             12.6
Year            2016.2
dtype: float64

#### What the average User Rating of the first 5 rows?
* answer: 4.7 

In [20]:
mode_user = bestsellers.mode(numeric_only=True)
mode_user

User Rating  Reviews  Price  Year
0           4.8   8580.0    8.0  2009
1           NaN      NaN    NaN  2010
2           NaN      NaN    NaN  2011
3           NaN      NaN    NaN  2012
4           NaN      NaN    NaN  2013
5           NaN      NaN    NaN  2014
6           NaN      NaN    NaN  2015
7           NaN      NaN    NaN  2016
8           NaN      NaN    NaN  2017
9           NaN      NaN    NaN  2018
10          NaN      NaN    NaN  2019

#### What User Review score appeared the most?
* answer: 4.8

In [21]:
total_reviews = bestsellers.sum(numeric_only=True)
total_reviews

User Rating       2540.1
Reviews        6574305.0
Price             7205.0
Year           1107700.0
dtype: float64

#### What is the total (sum) of all the values in the Reviews column?
* answer: 6574305

In [24]:
bestsellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         550 non-null    object 
 1   Author       550 non-null    object 
 2   User Rating  550 non-null    float64
 3   Reviews      550 non-null    int64  
 4   Price        550 non-null    int64  
 5   Year         550 non-null    int64  
 6   Genre        550 non-null    object 
dtypes: float64(1), int64(3), object(3)
memory usage: 30.2+ KB


In [26]:
autors = bestsellers.describe(include=["object"])
autors

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   

              Genre  
count           550  
unique            2  
top     Non Fiction  
freq            310

#### How many different authors are featured in the dataset?
* answer: 248

#### Which author wrote the most number of books on the list? How many did they write?
* answer: Jiff Kinney wrote 12 books 